python/pyspark/sql/connect/dataframe.py

```diff
@@ -325,10 +325,14 @@ def drop(self, *cols: "ColumnOrName") -> "DataFrame":
             session=self._session,
         )
 
-    def filter(self, condition: Expression) -> "DataFrame":
-        return DataFrame.withPlan(
-            plan.Filter(child=self._plan, filter=condition), session=self._session
-        )
+    def filter(self, condition: Union[Expression, str]) -> "DataFrame":
+        # The expression can be a SQL string expression.
```

Comment:
Nit:

I think this comment is a bit redundant as the code below is self-explained.